clients/spark/core/src/main/scala/io/treeverse/gc/UncommittedGarbageCollector.scala

```diff
@@ -74,9 +83,12 @@ object UncommittedGarbageCollector {
           // in case of no uncommitted entries
           spark.emptyDataFrame.withColumn("physical_address", lit(""))
         }
+
+      // TODO: should filter by time too
```

Comment:
You don't really need to filter committed and uncommitted in the client side. It's enough to remove it from the dataDF to ensure these won't be deleted

Reply:
Yesterday I had a strong hunch that having the 3 DFs (physicals, committed, uncommitted) looking at different time frames may lead to undeleted objects in the optimized run. I can't come up with the example now. 
We must consider the grace period when running the uncommitted GC, i.e. next optimized run's time frame needs to start from the previous run beginning of the grace period. Does it make sense?

Removing these comments for now.